### PR TITLE
feat(elements): Initial SignIn state machine

### PR DESCRIPTION
**Pull request by tmilewski**
## Description

A very initial implementation of Elements SignIn State Machine.

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: .github/.cache-version [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # Update this file to invalidate **all** cache on GitHub actions (use sparingly)
 
-version: v2
+version: v3
```
Comment (BRKalow) [inline .github/.cache-version:3]: ❓what prompted this?

File: packages/elements/examples/nextjs/app/globals.css [hunk 2]
```diff
@@ -18,10 +18,4 @@
 
 body {
   color: rgb(var(--foreground-rgb));
-  background: linear-gradient(
-      to bottom,
-      transparent,
-      rgb(var(--background-end-rgb))
-    )
-    rgb(var(--background-start-rgb));
```
Comment (BRKalow) [inline packages/elements/examples/nextjs/app/globals.css:26]: 🙏

File: .github/.cache-version (hunk 1 above)
Comment (tmilewski) [reply .github/.cache-version:3]: I'm trying to gauge why the linting process doesn't seem to want to find xstate

File: packages/elements/src/internals/machines/sign-in.actors.ts
Comment (BRKalow) [inline packages/elements/src/internals/machines/sign-in.actors.ts:1]: should this file be named `sign-in.actions`?

Comment (tmilewski) [reply packages/elements/src/internals/machines/sign-in.actors.ts:1]: In the eyes of XState, these are considered `actors`: https://stately.ai/docs/actors#frompromise

Comment (BRKalow) [reply packages/elements/src/internals/machines/sign-in.actors.ts:1]: Ah, interesting!

Comment (tmilewski) [reply packages/elements/src/internals/machines/sign-in.actors.ts:1]: Yup, actors can be long-lived, actions are a little more fire-and-forget.

**Review by BRKalow — APPROVED** [review]

**Pull request merged**